test/e2e/destroycluster/destroy_test.go

```diff
@@ -33,63 +42,179 @@ func TestDestroyCluster(t *testing.T) {
 	cd := common.MustGetClusterDeployment()
 	logger = logger.WithField("cluster", fmt.Sprintf("%s/%s", cd.Namespace, cd.Name))
 	if !cd.Spec.Installed {
-		logger.Warn("Cluster deployment was not installed")
+		logger.Warn("Cluster deployment did not finish installing")
 	}
 
 	fail := failTestFunc(t, logger)
 
-	c := common.MustGetClient()
+	c := common.MustGetHiveClient()
 	logger.Info("Deleting cluster deployment")
-	err := c.Delete(context.TODO(), cd)
+	err := c.HiveV1().ClusterDeployments(cd.Namespace).Delete(context.TODO(), cd.Name, metav1.DeleteOptions{})
 	if err != nil {
-		fail("Failed to delete cluster: %v", err)
+		fail("Failed to delete cluster deployment: %v", err)
 	}
 
+	go monitorDeprovisionJob(cd, logger)
+
 	logger.Info("Waiting for cluster deployment to go away")
-	waitForClusterDeploymentToGoAway(cd, fail)
+	err = waitForClusterDeploymentToGoAway(cd)
+	if err != nil {
+		fail("Failed to wait for cluster deployment to go away: %v", err)
+	}
 	logger.Info("Cluster deployment has been removed")
 
 	// Ensure that no dnszones owned by the clusterdeployment are left over
-	dnsZoneList := &hivev1.DNSZoneList{}
-	c.List(context.TODO(), dnsZoneList,
-		client.InNamespace(cd.Namespace),
-		client.MatchingLabels{
+	zoneLabelSelector, err := metav1.LabelSelectorAsSelector(&metav1.LabelSelector{
+		MatchLabels: map[string]string{
 			constants.DNSZoneTypeLabel:           constants.DNSZoneTypeChild,
 			constants.ClusterDeploymentNameLabel: cd.Name,
-		})
+		},
+	})
+	if err != nil {
+		fail("Failed to create a dns zone label selector: %v", err)
+	}
+	dnsZoneList, err := c.HiveV1().DNSZones(cd.Namespace).List(context.TODO(), metav1.ListOptions{
+		LabelSelector: zoneLabelSelector.String(),
+	})
+	if err != nil {
+		fail("Failed to get dns zone list: %v", err)
+	}
 	if len(dnsZoneList.Items) > 0 {
 		fail("At least one DNSZone still exists after clusterdeployment deletion")
 	}
 	logger.Info("No DNS Zones related to the cluster exist")
 }
 
-func waitForClusterDeploymentToGoAway(cd *hivev1.ClusterDeployment, fail func(string, ...interface{})) {
+var writeLogOnce sync.Once
+
+// monitor deprovision job will look for a deprovision job and save its log
+func monitorDeprovisionJob(cd *hivev1.ClusterDeployment, logger *log.Entry) {
+	if cd.Spec.ClusterMetadata == nil {
+		logger.Warn("Cluster metadata was never set, not monitoring uninstall job")
+		return
+	}
+
+	cfg := common.MustGetConfig()
+	gvk, err := apiutil.GVKForObject(&batchv1.Job{}, scheme.Scheme)
+	if err != nil {
+		logger.WithError(err).Error("Cannot get GVK for Job")
+		return
+	}
+	restClient, err := apiutil.RESTClientForGVK(gvk, cfg, serializer.NewCodecFactory(scheme.Scheme))
+	if err != nil {
+		logger.WithError(err).Error("Cannot get RESTClient for Job")
+		return
+	}
```

Comment:
Is there any benefit to getting an untyped rest client rather than the typed client?
```
import batchv1client "k8s.io/client-go/kubernetes/typed/batch/v1"
```

```suggestion
	batchClient, err := batchv1client.NewForConfig(cfg)
	if err != nil {
		logger.WithError(err).Error("cannot create batch client")
		return
	}
	restClient := batchClient.RESTClient() 
```

Reply:
no, I started with the generic client, but eventually changed, will fix